Signal/src/views/OWSAnyTouchGestureRecognizer.h

```diff
@@ -0,0 +1,16 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+// This custom GR can be used to detect touches when they
+// begin in a view.  In order to honor touch dispatch, this
+// GR will ignore touches that:
+//
+// * Are not single touches.
+// * Are not in the view for this GR.
+// * Are inside a visible, interaction-enabled subview.
+@interface OWSAnyTouchGestureRecognizer : UIGestureRecognizer
```

Comment:
We use a custom GR in order to be more responsive, specifically to recognize as soon as the user taps the screen, not when a "tap" is completed.